### PR TITLE
[master] Update unpackdiskimage to support rhcos fcp image

### PR DESCRIPTION
**Pull request by PanYaLian**
FCP, WWPN and LUN are required for copying rhcos fcp image

update unpackdiskimage to use FCP, WWPN and LUN to identify the device where the rhcos fcp images will be copied to

Signed-off-by: ylpan <ylpan@cn.ibm.com>

### CODE REVIEW

**Comment by PanYaLian** [comment]
[ylpan@oc5611624512 python-zvm-sdk]$ tox -epy27
py27 develop-inst-noop: /home/ylpan/mywork/CIC/python-zvm-sdk
py27 installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support,alabaster==0.7.12,attrs==19.3.0,Babel==2.8.0,certifi==2019.11.28,chardet==3.0.4,configparser==4.0.2,contextlib2==0.6.0.post1,docutils==0.16,extras==1.0.0,fixtures==3.0.0,funcsigs==1.0.2,functools32==3.2.3.post2,idna==2.9,imagesize==1.2.0,importlib-metadata==1.5.0,Jinja2==2.11.1,jsonschema==3.2.0,linecache2==1.0.0,MarkupSafe==1.1.1,mock==3.0.5,netaddr==0.7.19,packaging==20.1,pathlib2==2.3.5,pbr==5.4.4,Pygments==2.5.2,PyJWT==1.7.1,pyparsing==2.4.6,pyrsistent==0.15.7,python-mimeparse==1.6.0,python-subunit==1.3.0,pytz==2019.3,PyYAML==5.3,repoze.lru==0.7,requests==2.23.0,Routes==2.4.1,scandir==1.10.0,six==1.14.0,snowballstemmer==2.0.0,Sphinx==1.8.5,sphinxcontrib-websupport==1.1.2,testtools==2.3.0,traceback2==1.4.0,typing==3.7.4.1,unittest2==1.1.0,urllib3==1.25.8,WebOb==1.8.6,zipp==1.2.0,-e git+git@github.com:PanYaLian/python-zvm-sdk.git@7ff2e1138ec2cabbff2820eeaf06efd4fcb28106#egg=zVMCloudConnector
py27 run-test-pre: PYTHONHASHSEED='1619301287'
py27 run-test: commands[0] | python -m unittest discover -v -s /home/ylpan/mywork/CIC/python-zvm-sdk/zvmsdk/tests/unit
test_parse_normal (test_volumeop.TestFCP) ... ok
test_parse_npiv (test_volumeop.TestFCP) ... ok
test_add_fcp_for_assigner (test_volumeop.TestFCPManager) ... ok
test_expand_fcp_list_normal (test_volumeop.TestFCPManager) ... ok
test_expand_fcp_list_with_dash (test_volumeop.TestFCPManager) ... ok
test_expand_fcp_list_with_normal_plus_2dash (test_volumeop.TestFCPManager) ... ok
test_expand_fcp_list_with_normal_plus_dash (test_volumeop.TestFCPManager) ... ok
test_find_and_reserve_fcp_exception (test_volumeop.TestFCPManager) ... ok
test_find_and_reserve_fcp_new (test_volumeop.TestFCPManager) ... ok
................
test_get_image_performance_info_not_exist (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_nic_info (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_power_state (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_unpackdiskimage_cmd_rhcos (test_smtclient.SDKSMTClientTestCases) ... ok
**test_get_unpackdiskimage_cmd_rhcos_SCSI (test_smtclient.SDKSMTClientTestCases) ... ok**
test_get_unpackdiskimage_cmd_rhcos_skipcopy (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_user_console_output (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_user_console_output_request_failed (test_smtclient.SDKSMTClientTestCases) ... ok
test_get_user_direct (test_smtclient.SDKSMTClientTestCases) ... ok
...................
test_private_update_cpumem_data_cache_disabled (test_monitor.SDKMonitorTestCase) ... ok
test_private_update_cpumem_data_cache_enabled (test_monitor.SDKMonitorTestCase) ... ok
test_private_update_cpumem_data_cache_not_in_namelist (test_monitor.SDKMonitorTestCase) ... ok
test_private_update_nic_data (test_monitor.SDKMonitorTestCase) ... ok
test_private_update_nic_data_cache_disabled (test_monitor.SDKMonitorTestCase) ... ok
test_convert_to_mb (test_utils.ZVMUtilsTestCases) ... ok
test_get_namelist (test_utils.ZVMUtilsTestCases) ... ok

----------------------------------------------------------------------
Ran 629 tests in 1.386s

OK (skipped=3)
py27 run-test: commands[1] | python -m unittest discover -v -s /home/ylpan/mywork/CIC/python-zvm-sdk/smtLayer/tests/unit
test_getDiskPoolSpace3390 (test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace3390Unknown (test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336 (test_getHost.SMTGetHostTestCase) ... ok
test_getDiskPoolSpace9336Unknown (test_getHost.SMTGetHostTestCase) ... ok
test_getVM_directory_py3 (test_vmUtils.SMTvmUtilsTestCase) ... ok
test_getReservedMemSize (test_makeVM.SMTMakeVMTestCase) ... ok
test_getReservedMemSize_equal_size (test_makeVM.SMTMakeVMTestCase) ... ok
test_getReservedMemSize_gap_G (test_makeVM.SMTMakeVMTestCase) ... ok
test_getReservedMemSize_invalid_suffix (test_makeVM.SMTMakeVMTestCase) ... ok
test_getReservedMemSize_max_less_than_initial (test_makeVM.SMTMakeVMTestCase) ... ok

----------------------------------------------------------------------
Ran 10 tests in 0.003s

OK
___________________________________________________________ summary ____________________________________________________________
  py27: commands succeeded
  congratulations :)

File: zthin-parts/zthin/bin/unpackdiskimage [hunk 1]
```diff
@@ -70,11 +70,6 @@ function parseArgs {
     getPositionalArg 5 diskType
     getPositionalArg 6 nicID
     getPositionalArg 7 ipConfig
-
-    # Set fcp channel and make channel lower case
-    if [[ $diskType = 'SCSI' ]]; then
-      fcpChannel=$(echo ${channelID} | tr '[:upper:]' '[:lower:]')
-    fi
   elif [[ ${#args[@]} == 6 ]]; then
```
Comment (bjhuangr) [inline zthin-parts/zthin/bin/unpackdiskimage]: it would be better if we can check #args == 6, then 7

File: zthin-parts/zthin/bin/unpackdiskimage [hunk 2]
```diff
@@ -1171,34 +1186,19 @@ EOF
     # @Description:
     #   Deploy the specified SCSI disk image.
```
Comment (bjhuangr) [inline zthin-parts/zthin/bin/unpackdiskimage]: could u please help to update the description of this function?

File: zthin-parts/zthin/bin/unpackdiskimage [hunk 3]
```diff
@@ -1171,34 +1186,19 @@ EOF
...
-    wwpn=$(ls /dev/disk/by-path/ |grep -E ccw-0.0.${fcpChannel}-zfcp-|cut -d "-" -f 4 |tail -n 1 |cut -d ":" -f 1)
-    lun=$(ls /dev/disk/by-path/ |grep -E ccw-0.0.${fcpChannel}-zfcp-|cut -d "-" -f 4 |tail -n 1 |cut -d ":" -f 2)
+
     local DEST_DEV=`ls -al /dev/disk/by-path/ccw-0.0.${fcpChannel}-zfcp-${wwpn}:${lun} | tr '/' ' ' | awk '{print \$NF}'`
     DEST_DEV=/dev/$DEST_DEV
 
-    inform "Copy partition..."
-    dd if=${imageFile} bs=512 iflag=fullblock of="${DEST_DEV}" status=progress
+    if [[ $skipcopy != 'YES' ]]; then 
+        inform "Copy partition..."
```
Comment (bjhuangr) [inline zthin-parts/zthin/bin/unpackdiskimage]: dump to the whole disk? please update the message

**Pull request merged**